### ISSUE TITLE
Help needed: Is my label alignment correct?

### ISSUE THREAD

**Issue by teinhonglo** [slp-rl/WhiStress#1]
Hi,

Thank you for sharing this excellent work!

I’m currently implementing my own training pipeline based on your `evaluation_example.py`, and wrote a custom [`train.py`](https://github.com/teinhonglo/WhiStress/blob/main/train.py). While my training achieves an F1 score around **0.93**, using `evaluation_example.py` for inference only gives **\~0.26** F1.

I suspect the issue might come from my `preprocess()` logic for aligning token-level labels from word-level stress annotations. Here's the function I'm using:

```python
def preprocess(example, model):
    # 1. word-level binary label
    binary = compute_stress_binary(example["transcription"], example["emphasis_indices"])
    example["stress_pattern"] = {"binary": binary}

    # 2. prepare audio
    array = prepare_audio(audio=example["audio"])
    example["audio_input"] = {
        "array": array,
        "sampling_rate": 16000
    }

    # 3. tokenize transcription
    transcription = example["transcription"]
    tokenized = model.processor.tokenizer(
        transcription,
        return_tensors="pt",
        truncation=True,
        max_length=50
    )

    input_ids = tokenized["input_ids"][0]
    decoded_tokens = [
        model.processor.tokenizer.decode([tid], skip_special_tokens=False)
        for tid in input_ids
    ]

    # 4. align word-level binary to token-level
    token_labels = []
    word_idx = 0
    for token in decoded_tokens:
        if token in model.processor.tokenizer.all_special_tokens:
            token_labels.append(-100)
            continue

        if token.startswith(" "):  # new word
            label = binary[word_idx] if word_idx < len(binary) else 0
            token_labels.append(label)
            word_idx += 1
        else:  # subword
            label = binary[word_idx - 1] if word_idx - 1 < len(binary) else 0
            token_labels.append(label)

    # 5. LEFT SHIFT → align label with logits[t] = prediction of token[t+1]
    token_labels = token_labels[1:] + [-100]

    # 6. store results
    example["decoder_input_ids"] = input_ids
    example["labels_head"] = token_labels
    return example
```

My question is:

Is this label alignment strategy consistent with your original training setup?
Or is there something wrong that might explain the huge F1 drop during evaluation?

Any suggestions would be greatly appreciated!

Best regards,
Tien-Hong Lo
